ilastik/applets/batchProcessing/batchProcessingGui.py

```diff
@@ -211,109 +215,44 @@ def initAppletDrawerUi(self):
-            compute_in_2d |= opFeatureSelection.ComputeIn2d.value[col]
-
-        workflow.shell.projectManager.saveProject()
-
-        with io.BytesIO() as buf:
-            with h5py.File(buf) as dest:
-
-                def partial_copy(name, obj):
-                    if isinstance(obj, h5py.Group):
-                        dest.create_group(name).attrs.update(obj.attrs)
-                    elif isinstance(obj, h5py.Dataset) and name.startswith("Input Data/local_data"):
-                        dest.create_dataset_like(name, obj).attrs.update(obj.attrs)
-                    else:
-                        dest.copy(obj, name)
-
-                project_file.visititems(partial_copy)
-
-            data = buf.getvalue()
+        opDataSelection = workflow.dataSelectionApplet.topLevelOperator
+        opFeatureSelection = workflow.featureSelectionApplet.topLevelOperator
+
+        workflow.shell.projectManager.saveProject()  # Ensure that project_file is in sync.
+        project_file = opDataSelection.ProjectFile.value
+
+        payload = hbp.Payload(
+            token=token,
+            project=hbp.serialize_project_file(project_file),
+            filename=os.path.basename(project_file.filename),
+            num_channels=hbp.num_channels(opDataSelection),
+            min_block_size=hbp.min_block_size(opFeatureSelection),
+            compute_in_2d=hbp.compute_in_2d(opFeatureSelection),
+            workflow=self._hbp_category,
+        )
 
         try:
-            file_response = requests.post(
-                ilastik.config.cfg["hbp"]["upload_file_url"],
-                data=data,
-                headers={
-                    "Authorization": f"Token {token}",
-                    "Content-Disposition": f'attachment; filename="{filename}"',
-                },
-                timeout=10,
-            )
-            file_response.raise_for_status()
-            file_json = file_response.json()
-
-            project_response = requests.post(
-                ilastik.config.cfg["hbp"]["create_project_url"],
-                json={
-                    "file": file_json["url"],
-                    "num_channels": num_channels,
-                    "min_block_size_z": 0 if compute_in_2d else min_block_size,
-                    "min_block_size_y": min_block_size,
-                    "min_block_size_x": min_block_size,
-                },
-                headers={"Authorization": f"Token {token}"},
-                timeout=10,
-            )
-            project_response.raise_for_status()
-            project_json = project_response.json()
-
-            webbrowser.open_new_tab(project_json["html_url"])
-
-        except requests.exceptions.RequestException as e:
-            QMessageBox.critical(self, "Network Error", str(e))
+            webpage_url = hbp.send_payload(payload)
+            webbrowser.open(webpage_url)
+        except Exception as e:
+            QMessageBox.critical(self, "Exception", f"<pre>{e}</pre>")
```

Comment:
The message could probably be more verbose here, like "Project upload not successful..."